tests/integration/targets/test_moving_objects_between_buckets/tasks/main.yml

```diff
@@ -0,0 +1,124 @@
+        bucket: "{{ bucket_name_src }}"
+        object: /template/test.txt
+        content: "{{ lookup('file', 'test.txt') }}"
+        mode: put
+
+    - name: Put object (python) in source bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket_name_src }}"
+        object: test.py
+        content: "{{ lookup('file', 'test.py') }}"
+        mode: put
+
+    - slurp:
+        src: "{{ role_path }}/files/test.png"
+      register: put_binary
+
+    - name: Put object (image) in source s3 bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket_name_src }}"
+        object: put-binary.bin
+        content_base64: "{{ put_binary.content }}"
+        mode: put
+
+    # Moving one object between buckets =======================================================
+    - name: 
+      ansible.builtin.include_role:
+        name: cloud.aws_ops.moving_objects_between_buckets
+      vars:
+        source_bucket: "{{ bucket_name_src }}"
+        dest_bucket: "{{ bucket_name_dest }}"
+        source_bucket_object_prefix: "template"
+
+    - name: Obtaining source bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket_name_src }}"
+        mode: list
+      register: bucket_src_info
+
+    - name: Check the source s3 bucket remaining objects
+      ansible.builtin.assert:
+        that:
+        - "bucket_src_info.s3_keys[0] == 'put-binary.bin'"
+        - "bucket_src_info.s3_keys[1] == 'test.py'"
+        
+    - name: Obtaining destination bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket_name_dest }}"
+        mode: list
+      register: bucket_dest_info
+
+    - name: Check the destination s3 bucket obtained one object
+      ansible.builtin.assert:
+        that:
+        - "bucket_dest_info.s3_keys[0] == 'template/test.txt'"
+    
+    # Moving all objects between buckets =======================================================
+    - name: 
+      ansible.builtin.include_role:
+        name: cloud.aws_ops.moving_objects_between_buckets
+      vars:
+        source_bucket: "{{ bucket_name_src }}"
+        dest_bucket: "{{ bucket_name_dest }}"
+
+    - name: Obtaining source bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket_name_src }}"
+        mode: list
+      register: bucket_src_info
+
+    - name: Check the source s3 bucket is empty
+      ansible.builtin.assert:
+        that:
+        - "bucket_src_info.s3_keys == []"
+        
+    - name: Obtaining destination bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket_name_dest }}"
+        mode: list
+      register: bucket_dest_info
+
+    - name: Check the destination s3 bucket received all objects
+      ansible.builtin.assert:
+        that:
+        - "bucket_dest_info.s3_keys[0] in ['put-binary.bin','template/test.txt', 'test.py']"
+        - "bucket_dest_info.s3_keys[1] in ['put-binary.bin','template/test.txt', 'test.py']"
+        - "bucket_dest_info.s3_keys[2] in ['put-binary.bin','template/test.txt', 'test.py']"
+
+  always:
```

Comment:
You should also include `ignore_errors: true` for each task in the always block.